apis/chef/src/main/java/org/jclouds/chef/domain/Client.java

```diff
@@ -73,27 +80,34 @@ public Builder isValidator(boolean validator) {
       }
 
       public Client build() {
-         return new Client(certificate, orgname, clientname, name, validator, privateKey);
+         return new Client(certificate, orgname, clientname, name, validator, privateKey, publicKey);
       }
    }
 
    private final X509Certificate certificate;
    @SerializedName("private_key")
    private final PrivateKey privateKey;
+   @SerializedName("public_key")
+   private final PublicKey publicKey;
    private final String orgname;
    private final String clientname;
    private final String name;
    private final boolean validator;
 
-   @ConstructorProperties({ "certificate", "orgname", "clientname", "name", "validator", "private_key" })
+   @ConstructorProperties({ "certificate", "orgname", "clientname", "name", "validator", "private_key", "public_key"})
    protected Client(X509Certificate certificate, String orgname, String clientname, String name, boolean validator,
-         @Nullable PrivateKey privateKey) {
+         @Nullable PrivateKey privateKey, PublicKey publicKey) {
```

Comment:
If the public key is not returned in Chef Server versions < 12 it should be marked as nullable. Otherwise the deserialization of an object missing the public key will fail.

If this is the case, then could you add a test to verify this? In the test class you updated, keep a test for a client without the public key and one for a client that has it set.